### PR TITLE
Add `var.managed_iam_policy_enabled` to use a managed IAM policy or inline IAM policy

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Add `var.managed_iam_policy` to use a managed IAM policy or inline IAM policy
* Maintains backwards compatibility by defaulting this to `true`

## why
* Most of the time people do not need a managed IAM policy which are useful when an IAM policy should be reused.
* If a single purpose EKS IAM role is created, the IAM policy most likely does not need to be reused and so the `var.managed_iam_policy` can be set to `false` to switch to an inline IAM policy.

## references
* N/A

### CODE REVIEW

**Review by Gowiem — COMMENTED** [review]
Looks like this will have conflicts with #16... let's wait til that lands to get this merged since that had been around for a bit?

File: variables.tf [hunk 1]
```diff
@@ -28,3 +28,9 @@ variable "eks_cluster_oidc_issuer_url" {
   type        = string
   description = "OIDC issuer URL for the EKS cluster (initial \"https://\" may be omitted)"
 }
+
+variable "managed_iam_policy" {
```
Comment (osterman) [inline variables.tf]:
```suggestion
variable "managed_iam_policy_enabled" {
```

**Review by osterman — DISMISSED** [review]

**Comment by sebastianmacarescu** [comment]
Can this be merged?

**Comment by Gowiem** [comment]
@nitrocode friendly ping -- Looks like this lost traction and has conflicts now, but still seems valid. Want to get it updated and I'll give you a final review?

**Comment by ryanobjc** [comment]
Ah yes I was looking for this feature today... would be nice!

**Pull request closed without merging**